clickhouse/base/sslsocket.cpp

```diff
@@ -181,6 +184,33 @@ SSLSocket::SSLSocket(const NetworkAddress& addr, const SSLParams & ssl_params, S
     }
 }
 
+SSLSocketFactory::~SSLSocketFactory() {}
+
+#if defined(WITH_OPENSSL)
+std::unique_ptr<Socket> SSLSocketFactory::doConnect(const ClientOptions& opts,
+                                                    const NetworkAddress& address) {
+    std::unique_ptr<SSLContext> ssl_context;
```

Comment:
IMO it would make sense to store `SSLContext` in a `SSLSocketFactory` and create it only once: in constructor.